packaging/linux/post_install.sh

```diff
@@ -32,6 +33,68 @@ make_mountpoint() {
   fi
 }
 
+# We use machinectl here because systemctl commands do not work in either a su shell or with systemd-run --uid.
+exec_as() {
+    user=$1
+    shift
+    machinectl -q shell "$user"@ "$@"
+}
+
+# Exits with 0 iff the given user is running the service with systemd
+systemd_unit_active_for() {
+    user=$1
+    service=$2
+
+    # exit with 1 if command fails (for example, if not using systemd)
+    active_status="$(exec_as "$user" /usr/bin/systemctl --user is-active "$service")" || return "$?"
+
+    # exit code lost by machinectl, so grep stdout
+    # use -w so "active" needle doesn't match "inactive"
+    echo "$active_status" | grep -wq "active"
+}
+
+systemctl_restart_as() {
+    user=$1
+    service=$2
+    echo "Restarting $service for $user..."
+    exec_as "$user" /usr/bin/systemctl --user restart "$service"
+}
+
+systemd_safe_restart_services() {
+    pids="$(pidof /usr/bin/keybase)"
+    while read -r pid; do
+        user="$(ps -o user= -p "$pid")"
+
+        if systemd_unit_active_for "$user" "kbfs.service"; then
+            # KBFS running, need to check mounts before proceeding
+
+            # Don't use `mount(8)` to find kbfs mounts in case there are non-kbfs fuse systems
+            mount="$(exec_as "$user" /usr/bin/keybase config get -b mountdir | tr -d '\r')"
+            uses="$(exec_as "$user" /usr/bin/lsof "$mount")"
```

Comment:
can't use exit code because lsof returns zero when something is found, and 1 when either nothing was found or there's an error.

Reply:
What do you think about changing the logic to "if lsof successfully returns in-use files, don't restart. Otherwise, if lsof hits an error or there are no files in use, do restart". A bit more aggressive.

Reply:
Is the issue that the error will be present in `$uses`, so we'd still abort as below?

What are some cases where `lsof` returns an error?  Is it basically just the mount not existing?  In that case it seems fine to restart, so I think I agree with your proposal.

Reply:
That's one reason but there could be others in which case it might be unsafe, not sure. But this seems like a decent approach. I don't think the error would be in uses, it'd be in stdout. I just meant that that's why I used `-n $uses` instead of checking `$?`. That being said, the exit code isn't even being preserved by machinectl (I have a workaround for that though).

Reply:
Ah ok, cool.  This way is fine with me for now.  I thought you were proposing some change that you hadn't coded yet.